### DISCUSSION TITLE
Warnings that originate from external macro

### DISCUSSION THREAD

**Discussion by pacak** [mrcjkb/rustaceanvim#288] [Q&A] [answered]
First of all - thank you for reviving rust-tools, been using it and it mostly just works.

I'm getting warnings on a code generated by external substitution and proc macros, for example with anyhow I'm getting this:

```
    anyhow::bail!("Cannot locate the path to the asm file");     ■ replace return <expr>; with <expr> 
```

I went though the documentation of this crate and existing tickets and not sure how to disable those. Not my macro, can't do anything about it. rust-tools was quiet about it, I think the default behavior for this project should be the same.

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave:

I'll convert this to a Q&A discussion, since it's related to rust-analyzer, and not rustaceanvim directly.

You can enable/disable procMacro support in the [rust-analyzer configuration](https://rust-analyzer.github.io/manual.html#configuration). See also `:h rustaceanvim.config`.

**Reply by pacak** [reply]
Well, that didn't really help but it gave a good starting point. What I'm getting is probably related to this ticket - https://github.com/rust-lang/rust-analyzer/issues/16514

Disabling proc macro support replaces warnings with "macro not expanded" errors, I can probably disable those too, but they won't help in case of `anyhow` macro I listed above - it's not a proc macro.

I went to this bugtracker because it wasn't producing any errors with rust-tools and I couldn't find anything in my configuration to specifically disable it.

I guess I'll do a bit more research and going to complain to rust-analyzer bugtracker :)

**Reply by mrcjkb** [reply]
:thinking: it's strange that it didn't show up with rust-tools for you. Afaik, it doesn't apply any default rust-analyzer settings. Are you sure you were using the same version or rust-analyzer?
Since rust-tools used nvim-lspconfig under the hood, quite a few people were actually using a mason.nvim rust-analyzer installation (often without realising it).
That leads to some discrepancies, because the mason.nvim install often isn't built with the same toolchain as the rust project.

Glad you were able to find a discussion though. Best of luck :smile: 
And do let me know if you find a solution.

**Reply by pacak** [reply]
> Are you sure you were using the same version or rust-analyzer?

I think so. I usually update rustc the same day it releases and install rust-analyzer with rustup.

> And do let me know if you find a solution.

That's the plan.
